R/zigzagHMC.R

```diff
@@ -11,36 +11,38 @@
 #' @param upperBounds a d-dimensional vector specifying the upper bounds. `Inf` is accepted. 
 #' @param nutsFlg logical. If `TRUE` the No-U-Turn sampler will be used (Zigzag-NUTS).
 #' @param init a d-dimensional vector of the initial value. `init` must satisfy all constraints. If `init = NULL`, a random initial value will be used.
-#' @param step step size for Zigzag-HMC or Zigzag-NUTS (if `nutsFlg = TRUE`). Default value is the empirically optimal choice: sqrt(2)(lambda)^(-1/2) for Zigzag-HMC and 0.1(lambda)^(-1/2) for Zigzag-NUTS, where lambda is the minimal eigenvalue of the precision matrix.   
-#' @param rSeed random seed (default = 1).
+#' @param stepsize step size for Zigzag-HMC or Zigzag-NUTS (if `nutsFlg = TRUE`). Default value is the empirically optimal choice: sqrt(2)(lambda)^(-1/2) for Zigzag-HMC and 0.1(lambda)^(-1/2) for Zigzag-NUTS, where lambda is the minimal eigenvalue of the precision matrix.   
+#' @param seed random seed (default = 1).
+#' @param diagnosticMode logical. `TRUE` for also returning diagnostic information such as the stepsize used. 
 #'
-#' @return an (n + burnin)*d matrix of samples. The first `burnin` samples are from the user specified warm-up iterations.
+#' @return an nSample-by-d matrix of samples. If `diagnosticMode` is `TRUE`, a list with additional diagnostic information is returned. 
```

Comment:
```suggestion
#' @return an nSample x d matrix of samples. If `diagnosticMode` is `TRUE`, a list with additional diagnostic information is returned. 
```

to be consistent with `harmonicHMC`'s  documentation. Or if this method is preferred, would want to change the sentence in `harmonicHMC`

Reply:
I updated `harmonicHMC`'s doc b/c part of it already uses the "-by-" notation.  To be honest, though, the doc styles aren't consistent at all across `zigzagHMC` and `harmonicHMC`; I am ignoring it for now b/c they are just docs and are still readable.